Undefined reference to "caml_startup" when trying to run `dune utop`

In an ocaml-rs project, for example [ocaml-rust-starter](https://github.com/zshipko/ocaml-rust-starter), running `dune utop` gives the following error (while linking):
```
/usr/bin/ld: src/libocaml_rust_starter.a(ocaml_interop-4a398e80f3ca6ceb.ocaml_interop.8eq22uyz-cgu.12.rcgu.o): in function `_ZN3std4sync4once4Once9call_once28_$u7b$$u7b$closure$u7d$$u7d$17h7171c1b9a6fc7898E.llvm.3140953836603515050':
ocaml_interop.8eq22uyz-cgu.12:(.text._ZN3std4sync4once4Once9call_once28_$u7b$$u7b$closure$u7d$$u7d$17h7171c1b9a6fc7898E.llvm.3140953836603515050+0x3b): undefined reference to `caml_startup'
/usr/bin/ld: src/libocaml_rust_starter.a(ocaml_interop-4a398e80f3ca6ceb.ocaml_interop.8eq22uyz-cgu.12.rcgu.o): in function `_ZN4core3ops8function6FnOnce40call_once$u7b$$u7b$vtable.shim$u7d$$u7d$17h59471d095cabe6c5E.llvm.3140953836603515050':
ocaml_interop.8eq22uyz-cgu.12:(.text._ZN4core3ops8function6FnOnce40call_once$u7b$$u7b$vtable.shim$u7d$$u7d$17h59471d095cabe6c5E.llvm.3140953836603515050+0x3b): undefined reference to `caml_startup'
collect2: error: ld returned 1 exit status
```
![image](https://user-images.githubusercontent.com/7559560/120075777-543a9c80-c0ab-11eb-8641-672e1b14b46e.png)

Adding this function somewhere where the linker sees it, for example using the following code, "fixes" the issue and `utop` starts fine:
```rust
#[ocaml::func]
pub fn caml_startup() {}
```

OCaml version: 4.11.1
Rust version: nightly
GCC version: 10.2.0

The issue I have is actually different:

```
Undefined symbols for architecture x86_64:
  "_caml_startup", referenced from:
      std::sync::once::Once::call_once::_$u7b$$u7b$closure$u7d$$u7d$::had0dbd2626032528 (.llvm.15372601394740829014) in libmarlin_plonk_stubs.a(ocaml_interop-e80f53feb18288d8.ocaml_interop.ek75r3z1-cgu.12.rcgu.o)
      core::ops::function::FnOnce::call_once$u7b$$u7b$vtable.shim$u7d$$u7d$::h32e57e4f854a6975 (.llvm.15372601394740829014) in libmarlin_plonk_stubs.a(ocaml_interop-e80f53feb18288d8.ocaml_interop.ek75r3z1-cgu.12.rcgu.o)
     (maybe you meant: _caml_startup_code_exn, _caml_startup_aux , _caml_startup_code )
ld: symbol(s) not found for architecture x86_64
```

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -1,3 +1,8 @@
+## 0.22.1
+
+- Add `no-caml-startup` feature to allow `ocaml-rs` libraries to link
+  correctly when using `dune utop`
+
 ## 0.22.0
 
 - Allow `Value` to hold boxroot or raw value

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "ocaml"
-version = "0.22.0"
+version = "0.22.1"
 authors = ["Zach Shipko <zachshipko@gmail.com>"]
 readme = "README.md"
 keywords = ["ocaml", "rust", "ffi"]
@@ -14,7 +14,7 @@ edition = "2018"
 features = [ "without-ocamlopt", "derive", "link" ]
 
 [dependencies]
-ocaml-interop = "0.8"
+ocaml-interop = "^0.8.5"
 ocaml-sys = {path = "./sys", version = "0.22"}
 ocaml-boxroot-sys = {version = "0.2"}
 ocaml-derive = {path = "./derive", optional = true, version = "0.22"}
@@ -33,6 +33,7 @@ without-ocamlopt = [
 caml-state = ["ocaml-sys/caml-state"]
 no-std = ["cstr_core/alloc"]
 bigarray-ext = ["ndarray"]
+no-caml-startup = ["ocaml-interop/no-caml-startup"]
 
 [workspace]
 members = [

--- a/README.md
+++ b/README.md
@@ -52,6 +52,8 @@ By default, building `ocaml-sys` will invoke the `ocamlopt` command to figure ou
 - `OCAMLOPT` (default: `ocamlopt`) is the command that will invoke `ocamlopt`
 - `OCAML_VERSION` (default: result of `$OCAMLOPT -version`) is the target runtime OCaml version.
 - `OCAML_WHERE_PATH` (default: result of `$OCAMLOPT -where`) is the path of the OCaml standard library.
+- `OCAML_INTEROP_NO_CAML_STARTUP` (default: unset) can be set when loading an `ocaml-rs` library into an OCaml
+  bytecode runtime (such as `utop`) to avoid linking issues with `caml_startup`/`caml_shutdown`
 
 If both `OCAML_VERSION` and `OCAML_WHERE_PATH` are present, their values are used without invoking `ocamlopt`. If any of those two env variables is undefined, then `ocamlopt` will be invoked to obtain both values.
 

--- a/README.md
+++ b/README.md
@@ -53,7 +53,7 @@ By default, building `ocaml-sys` will invoke the `ocamlopt` command to figure ou
 - `OCAML_VERSION` (default: result of `$OCAMLOPT -version`) is the target runtime OCaml version.
 - `OCAML_WHERE_PATH` (default: result of `$OCAMLOPT -where`) is the path of the OCaml standard library.
 - `OCAML_INTEROP_NO_CAML_STARTUP` (default: unset) can be set when loading an `ocaml-rs` library into an OCaml
-  bytecode runtime (such as `utop`) to avoid linking issues with `caml_startup`/`caml_shutdown`
+  bytecode runtime (such as `utop`) to avoid linking issues with `caml_startup`
 
 If both `OCAML_VERSION` and `OCAML_WHERE_PATH` are present, their values are used without invoking `ocamlopt`. If any of those two env variables is undefined, then `ocamlopt` will be invoked to obtain both values.
 

--- a/derive/src/derive.rs
+++ b/derive/src/derive.rs
@@ -69,7 +69,7 @@ pub fn intovalue_derive(mut s: synstructure::Structure) -> proc_macro::TokenStre
         };
         let tag = *tag_ref;
         *tag_ref += 1;
-        let attrs = variant_attrs(&variant.ast().attrs);
+        let attrs = variant_attrs(variant.ast().attrs);
         if (attrs.floats || attrs.unboxed) && !is_record_like {
             panic!("ocaml cannot derive unboxed or float arrays for enums")
         }
@@ -137,7 +137,7 @@ pub fn fromvalue_derive(s: synstructure::Structure) -> proc_macro::TokenStream {
         } else {
             &mut unit_tag
         };
-        let attrs = variant_attrs(&variant.ast().attrs);
+        let attrs = variant_attrs(variant.ast().attrs);
         if (attrs.floats || attrs.unboxed) && !is_record_like {
             panic!("ocaml cannot derive unboxed records or float arrays for enums")
         }

